### PR TITLE
CB-9597 Adds implementation for PlatformApiPoly class according to PlatformApi spec

### PR DESCRIPTION
**Pull request by vladimir-kotikov**
This implements PlatformApiPoly class according to PlatformApi spec, which allows to:
  * create/update platform
  * execute platform's actions (build/run/add/update)
  * do a prepare (needed for CLI workflow only)
  * install/uninstall plugins

Other noticeable changes:
  * removes `getPlatformProject` and PlatformProject method/class in favor of PlatformApiPoly/getPlatformApi
  * make assets and js-modules installing/uninstalling through ActionStack
  * refactor configChanges to not require plugins_dir in constructor
  * moves mergeXml helper to xml-helpers

This should be used along with cordova-cli/platformApi branch: https://github.com/apache/cordova-cli/pull/220

### CODE REVIEW

**Comment by kamrik** [comment]
Looks good! and very clean.

One note about tests, jasmine spies seem to give you a lot of trouble. In the previous refactoring round I was doing, I found it very useful to entirely replace some tests that rely on spies too much with something based on filesystem fixtures with less spying.

File: cordova-lib/src/plugman/util/config-changes.js [hunk 1]
```diff
@@ -168,22 +167,18 @@ function remove_plugin_changes(plugin_name, plugin_id, is_top_level) {
     }
 
     // Remove from installed_plugins
-    if (is_top_level) {
-        delete platform_config.installed_plugins[plugin_id];
-    } else {
-        delete platform_config.dependent_plugins[plugin_id];
-    }
+    self.platformJson.addPlugin(pluginInfo.id, is_top_level);
```
Comment (vladimir-kotikov) [inline cordova-lib/src/plugman/util/config-changes.js]: FIXME: probably should be `removePlugin()`

**Comment by vladimir-kotikov** [comment]
Thanks for advice, Mark. I have some plans about tests update and definitely will try fixtures-based approach.

**Comment by stevengill** [comment]
Looks great Vlad!  I'm fine with you merging this in and us spending some time testing it on master to make sure it works with all of our platforms.

**Pull request merged**